src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -423,37 +426,25 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
             cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgs))
             return report
 
-        reqArgsNothandled = []
-        for reqArg in reqArgs:
-            if reqArg == 'RequestPriority':
-                validate_request_priority(reqArgs)
-                # must update three places: GQ elements, workload_cache and workload spec
-                self.gq_service.updatePriority(workload.name(), reqArgs['RequestPriority'])
-                workload.setPriority(reqArgs['RequestPriority'])
-                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgs['RequestPriority']))
-            elif reqArg == "SiteWhitelist":
-                workload.setSiteWhitelist(reqArgs["SiteWhitelist"])
-                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteWhitelist']))
-            elif reqArg == "SiteBlacklist":
-                workload.setSiteBlacklist(reqArgs["SiteBlacklist"])
-                cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteBlacklist']))
-            else:
-                reqArgsNothandled.append(reqArg)
-                cherrypy.log("Unhandled argument for no-status update: %s" % reqArg)
-
-        reqStatus = self.reqmgr_db_service.getRequestByNames(workload.name())[workload.name()]['RequestStatus']
-        cherrypy.log(f"CurrentRequest status: {reqStatus}")
-        if reqArgsNothandled:
+        try:
+            # Update all workload parameters based on the full reqArgs dictionary
+            workload.updateWorkloadArgs(reqArgs)
+        except WMWorkloadUnhandledException as ex:
+            # Handling assignment-approved arguments differently to avoid code duplication
             if reqStatus == 'assignment-approved':
                 cherrypy.log(f"Handling assignment-approved arguments differently!")
                 self._handleAssignmentStateTransition(workload, request_args, dn)
             else:
-                msg = "There were unhandled arguments left for no-status update: %s" % reqArgsNothandled
-                raise InvalidSpecParameterValue(msg)
+                msg = "There were unhandled arguments left for no-status update."
+                raise InvalidSpecParameterValue(msg) from ex
 
         # Commit the changes of the current workload object to the database:
         workload.saveCouchUrl(workload.specUrl())
 
+        # Commit all Global WorkQueue changes per workflow in a single go:
+        self.gq_service.updateElementsByWorkflow(workload.name(), reqArgs, status=['Available', 'Negotiating', 'Acquired'])
```

Comment:
hi @amaltaro  the line that I was talking about during the WMCore meeting is this one. It triggers a second call to `workload.updateWorkloadArgs` internally, similarly to what is done on line: [431](https://github.com/dmwm/WMCore/pull/12099/files#diff-120ee6838284a3d1c1799f511da7f147179d0a955f87d0da6fc8b58a8b66c794R431)  at `WMcore.ReqMgr.Service.Request`  from this same PR. 

The internal call is happening  at line [316](https://github.com/dmwm/WMCore/pull/12099/files#diff-d758570ce7f6baddaeefb0bdb4e2015b06e5b7772bf929248b686adc64286233R316) at `WMCore.Services.WorkQueue` again from the very same PR.

You have asked me to implement all the changes to both `reqMgr` (in this case through the `workload` object) and all `WQE`s  in a single push. It is now  possible. All that needs to happen, is to substitute the first from above with the second call from the current line. I have the code change prepared. Just let me know  if I should proceed with it or we should play it safe and we still do the action in two steps.